src/chapters/typescript/functions.rst

```diff
@@ -0,0 +1,115 @@
+Functions in TypeScript
+=======================
+
+When creating functions in TypeScript, we have many of the same options as in JavaScript.
+We can make anonymous functions, give it a different number of parameters and so on.
+However, when working with functions in TypeScript, we have to keep *type* in mind.
+
+Declaring Functions
+-------------------
+
+Named Functions
+^^^^^^^^^^^^^^^
+
+Let's take a look at a TypeScript function declaration.
+
+.. sourcecode:: typescript
+
+   function myFunction(a: number): number {
+      return a*2;
+   }
+
+Here you can see that we provided the type of the parameters and the type of the value that is returned.
```

Comment:
I would break this down more since its the first time they are seeing this kind of syntax.
like explicitly, 
"here, the function myFunction has one parameter, ``a``. ``a`` is of type ``number`` and is declared as such. If ``a`` were a string, the function declaration would look like ``myFunction(a: string)``. If myFunction took two parameters, it would be declared as ``myFunction(a: number, b: string)``. myFunction returns a value that is of type ``number``, if it returned an array of numbers, it would be declared as ..."

You get it, just a more fine-grained explanation of what they are looking at in the example could be valuable.

Reply:
I added some more explanation!